src/core/services-api/src/main/java/it/geosolutions/geostore/services/ResourceService.java

```diff
@@ -200,6 +200,29 @@ List<Resource> getResources(
             User authUser)
             throws BadRequestServiceEx, InternalErrorServiceEx;
 
+    /**
+     * @param filter
+     * @param page
+     * @param entries
+     * @param sortBy
+     * @param sortOrder
+     * @param includeAttributes
+     * @param includeData
+     * @return List<Resource>
+     * @throws BadRequestServiceEx
+     * @throws InternalErrorServiceEx
+     */
+    List<Resource> getResources(
+            SearchFilter filter,
+            Integer page,
+            Integer entries,
+            String sortBy,
+            String sortOrder,
+            boolean includeAttributes,
+            boolean includeData,
+            User authUser)
+            throws BadRequestServiceEx, InternalErrorServiceEx;
```

Comment:
* If you foresee further enhancements (e.g., multiple sort fields), consider a SearchRequest or ResourceQuery object to encapsulate all pagination + sorting + filtering parameters. It keeps the method signature clean and reduces the risk of “parameter bloat.”

* Document or validate that sortBy must match a valid entity field. Right now, you only do a sortBy != null && !sortBy.isBlank(), but if the user passes an invalid field, it may throw an exception deeper in the DAO layer.

Reply:
Created a parameter object for the resource search.
Documented in the javadoc the usage of the search method.